src/main/java/mekanism/client/gui/GuiMekanism.java

```diff
@@ -64,29 +70,51 @@ public void renderScaledText(String text, int x, int y, int color, int maxX) {
 
     @Override
     protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
-        super.drawGuiContainerForegroundLayer(mouseX, mouseY);
-        int xAxis = mouseX - (width - xSize) / 2;
-        int yAxis = mouseY - (height - ySize) / 2;
-        guiElements.forEach(element -> element.renderForeground(xAxis, yAxis));
+        int xAxis = mouseX - guiLeft;
+        int yAxis = mouseY - guiTop;
+        //Ensure that the GL color is white, as drawing rectangles, text boxes, or even text might have changed the color from
+        // what we assume it is at the start. This prevents any unintentional color state leaks. GlStateManager, will ensure that
+        // GL changes only get ran if it is not already the color we are assuming it is.
+        GLSMHelper.resetColor();
+        guiElements.forEach(element -> {

```

Comment:
Cleaned up how it is done for the background, (and also elements). I am going to leave the singular reset before the elements start looping here though in case other things get us into an unexpected state and because almost every single Gui we have for rendering foreground changes the color when displaying text in one way or another and that would be a lot of duplicate calls to resetColor.